src/planner/join_order/cardinality_estimator.cpp

```diff
@@ -93,12 +109,33 @@ static bool isPrimaryKey(const Expression& expression) {
     return ((PropertyExpression&)expression).isPrimaryKey();
 }
 
+static bool isSingleLabelledProperty(const Expression& expression) {
+    if (expression.expressionType != ExpressionType::PROPERTY) {
+        return false;
+    }
+    return expression.constCast<PropertyExpression>().isSingleLabel();
+}
+
 uint64_t CardinalityEstimator::estimateFilter(const LogicalPlan& childPlan,
     const Expression& predicate) {
     if (predicate.expressionType == ExpressionType::EQUALS) {
         if (isPrimaryKey(*predicate.getChild(0)) || isPrimaryKey(*predicate.getChild(1))) {
             return 1;
         } else {
+            KU_ASSERT(predicate.getNumChildren() >= 1);
+            if (isSingleLabelledProperty(*predicate.getChild(0))) {
+                auto& propertyExpr = predicate.getChild(0)->cast<PropertyExpression>();
+                auto tableID = propertyExpr.getSingleTableID();
+                if (nodeTableStats.contains(tableID)) {
+                    auto columnID = propertyExpr.getColumnID(
+                        *context->getCatalog()->getTableCatalogEntry(context->getTx(), tableID));
+                    if (columnID != INVALID_COLUMN_ID && columnID != ROW_IDX_COLUMN_ID) {
```

Comment:
I would wrap the nested if statements into a separate function canApplyTableStats.

Reply:
moved to separate function